### ISSUE TITLE
"Server '<ip> <host>' not found in known_hosts" when expend new segment

### ISSUE THREAD

**Issue by ourdark** [greenplum-db/gpdb#5148]
### Greenplum version or build
5.7.0
### OS version and uname -a
Linux <host> 3.10.0-514.el7.x86_64 #1 SMP Tue Nov 22 16:42:41 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

CentOS Linux release 7.3.1611 (Core)
### Expected behavior
gpssh-exkeys -e hostfile_exkeys -x new_hosts_file 
return print '[INFO] completed successfully' after correct pwd
[STEP 3 of 5] authorize current user on remote hosts
  ... send to <ip> <host>
  ***
  *** Enter password for <ip> <host>: 
### Actual behavior
[ERROR <ip> <host>] Server '<ip> <host>' not found in known_hosts
  ***
  *** Enter password for <ip> <host>: 
pwd prompt again
### Step to reproduce the behavior

- install gp with mdw,smdw and (sdw1-3) data segement(ip: 10.129.7.*)

- preparing sdw4 for expation with(ip: 10.129.10.*)

- add hosts to /etc/hosts for sdw4

- run gpssh-exkeys -e hostfile_exkeys -x new_hosts_file 

- type correct pwd for sdw4

**Comment by tramer-pivotal** [comment]
@ourdark  I cannot reproduce this issue, even with white space and newline in the new hostfile:

~~~
[gpadmin@master ~]$ cat hostfile
node1

[gpadmin@master ~]$ cat hostfile_new

node2


[gpadmin@master ~]$ wc -l hostfile_new
3 hostfile_new


[gpadmin@master ~]$ gpssh-exkeys -e hostfile -x hostfile_new
[STEP 1 of 5] create local ID and authorize on local host
  ... /home/gpadmin/.ssh/id_rsa file exists ... key generation skipped

[STEP 2 of 5] keyscan all hosts and update known_hosts file

[STEP 3 of 5] authorize current user on remote hosts
  ... send to node2
/home/gpadmin/.ssh/id_rsa
  ***
  *** Enter password for node2:
/home/gpadmin/.ssh/id_rsa
  ... send to node1
/home/gpadmin/.ssh/id_rsa

[STEP 4 of 5] determine common authentication file content

[STEP 5 of 5] copy authentication files to all remote hosts
  ... finished key exchange with node2
  ... finished key exchange with node1

[INFO] completed successfully
~~~

Does this issue persist when you use the -f option? Also, do you have any blank lines in your .ssh/known_hosts file? 

Is the hostname/IP for sdw4 correctly entered in /etc/hosts? 

Is there some other DNS service apart from /etc/hosts which might be resolving the hostname sdw4 incorrectly? (for example, can you remove sdw4 from the /etc/hosts file to see if this makes a difference? Can you confirm the ordering of your hosts routing is correct in `/etc/nsswitch.conf` ? )

**Comment by ourdark** [comment]
@tramer-pivotal 

This issue can be repro anytime in my environment. 

Here my though(might be the problem): the sdw4 in diff network segment and virtualized by VMware 14.

I can ping sdw4 on each host coresponding correct ip addr. also sdw4 can access any resource in range. I've tried remove sdw4 then add it back, nothing changed. I've tried remove sdw4 from know_hosts still not working. And .... i can ssh sdw4 passwordless on mdw, smdw(port 22 is working good).

The DNS is working fine, for some reason, i made sdw4 to be access internet, and sdw4 is online.

I'll try to move sdw4 to 10.129.7.*(instead of 10.129.10.*), if that works for me, i'll let you know.
